### ISSUE TITLE
Add support for string interpolation

### ISSUE THREAD

**Issue by AlanFoster** [iamsolankiamit/prettier-ruby#8]
Prettier ruby doesn't currently support string interpolation:

```ruby
"1 + 2 + 3 = #{1 + 2 + 3}"
```

I'm taking a look at this currently, I'll post if I get anywhere :+1:

**Comment by AlanFoster** [comment]
@iamsolankiamit I think I'm blocked on this one until there's a generic way of matching a given sexp with the underlying tokens. Unfortunately the generated sexp tree for `'\n'` and `"\n"` are the same, but Ruby evaluates them differently. Therefore when prettying strings the two examples should be treated very differently - which I believe also stops us from using the generic prettier `makeString` util.

I think the implementation detail for getting access to the underlying token from an sexp will be similar to the concept you mentioned here https://github.com/iamsolankiamit/prettier-ruby/pull/4#discussion_r192073528 - Let me know if you've had any luck investigating that :+1:

**Comment by iamsolankiamit** [comment]
Hey, I'll be working on the tokens part tomorrow. You can start with this once it get's done.

I'll keep you posted here.

BTW, this
```bash
ruby vendor/ruby/astexport.rb '"an interpolated string #{interpolate_this}!"'
```
returns the following JSON
```json
{
  "ast_type": "program",
  "body": [
    {
      "ast_type": "string_literal",
      "string_content": [
        {
          "ast_type": "@tstring_content",
          "content": "an interpolated string "
        },
        {
          "ast_type": "string_embexpr",
          "error": "Unhandled node: string_embexpr"
        },
        {
          "ast_type": "@tstring_content",
          "content": "!"
        }
      ]
    }
  ],
  "comments": [

  ]
}
```

I guess this should be enough for us to handle interpolated string. Let me know if you need any other help!

**Comment by iamsolankiamit** [comment]
Fixed in b76c548
